plugins/blotter.js

```diff
@@ -0,0 +1,87 @@
+const fsw = require('fs');
+const _ = require('lodash');
+const log = require('../core/log.js');
+const util = require('../core/util.js');
+const config = util.getConfig();
+const blotterConfig = config.blotter;
+
+var Blotter = function(done) {
+  _.bindAll(this);
+
+  this.time;
+  this.valueAtBuy = 0.0;
+  this.filename = blotterConfig.filename;
+  this.dateformat = blotterConfig.dateFormat;
+  this.timezone = blotterConfig.timezone;
+  this.headertxt = '';
+  this.outtxt = '';
+
+  this.done = done;
+  this.setup();
+};
+
+Blotter.prototype.setup = function(done) {
+
+  this.headertxt = "Date,Price,Amount,Side,Fees,Value,P&L,Notes\n";
+
+  fsw.readFile(this.filename, (err, _) => {
+    if (err) {
+      log.warn('No file with the name', this.filename, 'found. Creating new blotter file');
+      fsw.appendFileSync(this.filename, this.headertxt, encoding='utf8'); 
+    } 
+  });
+
+};
+
+Blotter.prototype.processTradeCompleted = function(trade) {
```

Comment:
Can't you rely on the roundtrip event? That way you don't have to calculate the pnl yourself. See here: https://gekko.wizb.it/docs/internals/events.html#roundtrip-event

Reply:
I just did. I didn't realize the roundtrip event had p&l until after I wrote up this plugin, although I practically had to rewrite the plugin to use the roundtrip event. The benefit is the roundtrip event was more accurate for when sell amount doesn't match buy amount.